cmd/nodecmd/create.go

```diff
@@ -0,0 +1,678 @@
+	sgInput := &ec2.DescribeSecurityGroupsInput{
+		GroupNames: []*string{
+			aws.String(sgName),
+		},
+	}
+
+	sg, err := ec2Svc.DescribeSecurityGroups(sgInput)
+	if err != nil {
+		if strings.Contains(err.Error(), "InvalidGroup.NotFound") {
+			return false, &ec2.SecurityGroup{}, nil
+		}
+		return false, &ec2.SecurityGroup{}, err
+	}
+	return true, sg.SecurityGroups[0], nil
+}
+
+func checkCertInSSHDir(certFilePath string) bool {
+	_, err := os.Stat(certFilePath)
+	return err == nil
+}
+
+func checkCurrentIPInSg(sg *ec2.SecurityGroup, currentIP string) (bool, bool) {
+	var ipInTCP bool
+	var ipInHTTP bool
+	for _, ip := range sg.IpPermissions {
+		if *ip.FromPort == 22 || *ip.FromPort == 9650 {
+			for _, cidrIP := range ip.IpRanges {
+				if strings.Contains(cidrIP.String(), currentIP) {
+					if *ip.FromPort == 22 {
+						ipInTCP = true
+					} else if *ip.FromPort == 9650 {
+						ipInHTTP = true
+					}
+					break
+				}
+			}
+		}
+	}
+	return ipInTCP, ipInHTTP
+}
+
+func setCloudCredentials(rootBody *hclwrite.Body, accessKey, secretKey, region string) error {
+	provider := rootBody.AppendNewBlock("provider", []string{"aws"})
+	providerBody := provider.Body()
+	providerBody.SetAttributeValue("access_key", cty.StringVal(accessKey))
+	providerBody.SetAttributeValue("secret_key", cty.StringVal(secretKey))
+	providerBody.SetAttributeValue("region", cty.StringVal(region))
+
+	return nil
+}
+
+func setKeyPair(rootBody *hclwrite.Body, keyName, certName string) {
+	tlsPrivateKey := rootBody.AppendNewBlock("resource", []string{"tls_private_key", "pk"})
+	tlsPrivateKeyBody := tlsPrivateKey.Body()
+	tlsPrivateKeyBody.SetAttributeValue("algorithm", cty.StringVal("RSA"))
+	tlsPrivateKeyBody.SetAttributeValue("rsa_bits", cty.NumberIntVal(4096))
+
+	keyPair := rootBody.AppendNewBlock("resource", []string{"aws_key_pair", "kp"})
+	keyPairBody := keyPair.Body()
+	keyPairBody.SetAttributeValue("key_name", cty.StringVal(keyName))
+	keyPairBody.SetAttributeTraversal("public_key", hcl.Traversal{
+		hcl.TraverseRoot{
+			Name: "tls_private_key",
+		},
+		hcl.TraverseAttr{
+			Name: "pk",
+		},
+		hcl.TraverseAttr{
+			Name: "public_key_openssh",
+		},
+	})
+
+	tfKey := rootBody.AppendNewBlock("resource", []string{"local_file", "tf-key"})
+	tfKeyBody := tfKey.Body()
+	tfKeyBody.SetAttributeValue("filename", cty.StringVal(certName))
+	tfKeyBody.SetAttributeTraversal("content", hcl.Traversal{
+		hcl.TraverseRoot{
+			Name: "tls_private_key",
+		},
+		hcl.TraverseAttr{
+			Name: "pk",
+		},
+		hcl.TraverseAttr{
+			Name: "private_key_pem",
+		},
+	})
+}
+
+func getIPAddress() (string, error) {
+	ipOutput, err := exec.Command("curl", "ipecho.net/plain").Output()
+	if err != nil {
+		return "", err
+	}
+	ipAddress := string(ipOutput)
+	if net.ParseIP(ipAddress) == nil {
+		return "", errors.New("invalid IP address")
+	}
+	return ipAddress, nil
+}
+
+func setSecurityGroup(rootBody *hclwrite.Body, ipAddress, securityGroupName string) {
```

Comment:
We will probably want to run all this by infra and security before we include this in a release. We should also audit a sample deployment.